### ISSUE TITLE
Feature Request: Elasticsearch Service Kibana with Cognito authentication

### ISSUE THREAD

**Issue by limitusus** [terraform-providers/terraform-provider-aws#4061]
AWS released Cognito authentication integration to Kibana.
https://forums.aws.amazon.com/ann.jspa?annID=5602

### Terraform Configuration Files

Sample configuration I suppose; may change

```hcl
resource "aws_elasticsearch_domain" "es-test" {
  # other configurations...

  kibana_authentication {
    cognito_region = "us-east-1"
    cognito_user_pool = ""
    cognito_identity_pool = ""
    iam_role = ""
  }
}
```

### Expected Behavior

Possible to set, update, and remove authentication configuration

### Actual Behavior

Not implemented yet

**Comment by bflad** [comment]
Initial support has been merged into master via #5346 and will release with version 1.30.0 of the AWS provider, later today. Great work, @aoggz!

**Comment by bflad** [comment]
This has been released in [version 1.30.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1300-august-02-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
